yarn-project/end-to-end/src/e2e_l1_to_l2_msg.test.ts

```diff
@@ -158,31 +156,44 @@ describe.skip('e2e_l1_to_l2_msg', () => {
     const deadline = 2 ** 32 - 1; // max uint32 - 1
 
     logger('Sending messages to L1 portal');
-    const returnedMessageKey = await tokenPortal.write.depositToAztec(
-      [l2TokenAddress, 100n, deadline, secretString],
-      {} as any,
-    );
-
-    const messageKeyFr = Fr.fromBuffer(Buffer.from(returnedMessageKey, 'hex'));
-
-    // Wait for the rollup to process the message
-    // TODO: not implemented
-
-    // Force the node to consume the message
-    // TODO: not implemented
+    const args = [ownerAddress.toString(), 100n, deadline, secretString] as const;
+    const { result: messageKeyHex } = await tokenPortal.simulate.depositToAztec(args, {
+      account: ethAccount.toString(),
+    } as any);
+    await tokenPortal.write.depositToAztec(args, {} as any);
+    const messageKey = Fr.fromString(messageKeyHex);
+
+    // Wait for the archiver to process the message
+    const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+    await delay(5000); /// waiting 1 second.
+
+    // send a transfer tx to force through rollup with the message included
+    const transferAmount = 1n;
+    const transferTx = contract.methods
+      .transfer(
+        transferAmount,
+        pointToPublicKey(await aztecRpcServer.getAccountPublicKey(ownerAddress)),
+        pointToPublicKey(await aztecRpcServer.getAccountPublicKey(receiver)),
+      )
+      .send({ from: accounts[0] });
+
+    await transferTx.isMined(0, 0.1);
+    const transferReceipt = await transferTx.getReceipt();
+
+    expect(transferReceipt.status).toBe(TxStatus.MINED);
 
+    logger('Consuming messages on L2');
     // Call the mint tokens function on the noir contract
     const mintAmount = 100n;
 
```

Comment:
question - should we add `expect()` on methods in the archiver? See if the archiver actually processed everything correctly (i.e. added to pending store and later added to confirmed store)?

Reply:
Sounds like a good to have, ill add this